src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -157,14 +157,20 @@ public Optional<FieldChange> setMonth(Month parsedMonth) {
         return setField(StandardField.MONTH, parsedMonth.getJabRefFormat());
     }
 
+    /**
+     * Retrieves the first resolved field or its alias from the provided OrFields object using Stream API.
+     *
+     * @param fields    The OrFields object containing a list of fields to be resolved.
+     * @param database  The BibDatabase used for resolving the field or its alias.
+     * @return          An Optional containing the first resolved field or alias if present,
+     *                  otherwise an empty Optional.
+     */
     public Optional<String> getResolvedFieldOrAlias(OrFields fields, BibDatabase database) {
-        for (Field field : fields.getFields()) {
-            Optional<String> value = getResolvedFieldOrAlias(field, database);
-            if (value.isPresent()) {
-                return value;
-            }
-        }
-        return Optional.empty();
+        return fields.getFields().stream()
+                     .map(field -> getResolvedFieldOrAlias(field, database))
+                     .filter(Optional::isPresent)
+                     .map(Optional::get)
+                     .findFirst();
     }
```

Comment:
As a side note: Although streams can under certain circumstances speed up the execution and make it more readable (so this change is ok as far as I can see), it makes it more complicated to debug, since the stacktrace is broken up by the jdk internal stream handling. @Siedlerchr should confirm this.

Reply:
Yes, it really depends on the context

Reply:
@Toro520 Here, speed comparisons would be interesting. - I bet that the current for loop is faster - or does the compiler optimize the stream that much so that it is equal? (in the stream version, ALL elements are mapped and then filtered, then mapped and then the first is taken - not sure if there is an optimizuation for that in the JDK)

Reply:
Thank you guys, I have learnt a lot from this, and this really raise my enthusiasm to look through entire code and make the better decision next time. But right now where should I start to solve the issue #9717?  I considered how to be like you guys to have the deep learning and understanding of the code and try to solve it. If at current state, first year study CS at ANU, what kind of assistance I could support. I know ,first of all, do not cause too much trouble should be the priority. 😅

Reply:
@Toro520 The issue is much about code reading. The real code won't be much. -- Try to understand how the main table displays data. I think, for crossref  one has to manipulate that thing so that not only the bibentry at hand, but also the cross-ref'd entry is taken into consideration.

Loom at the discussion at https://github.com/JabRef/jabref/pull/7754 - there was a first attempt in that direction.

(I do not know about the code exactly. -- If, let's say, within 5 hours, you don't have any clue (you can also ask here or at that PR), consider changing the issue)

Reply:
@k3KAW8Pnf7mkmdSMPHz27 Just for information: crossref is looked at by a student. Hope, it is manageable ^^

Reply:
@koppor Thanks so much. 👍

Reply:
> in the stream version, ALL elements are mapped and then filtered, then mapped and then the first is taken

Intermediate operations should be [lazily evaluated](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/stream/package-summary.html#:~:text=Laziness%2Dseeking.%20Many,are%20always%20lazy.)

> Laziness-seeking. Many stream operations, such as filtering, mapping, or duplicate removal, can be implemented lazily, exposing opportunities for optimization. For example, "find the first String with three consecutive vowels" need not examine all the input strings. Stream operations are divided into intermediate (Stream-producing) operations and terminal (value- or side-effect-producing) operations. Intermediate operations are always lazy.

I would use a flatMap rather than two filtering operations, but I don't think the performance improvement is significant.

> I bet that the current for loop is faster - or does the compiler optimize the stream that much so that it is equal?

I can't think of any case where a stream will be faster than a well-written for-loop. Streams are IMO well optimized, but so are for-loops. I'd vote to leave for-loops in performance-critical locations.

> @k3KAW8Pnf7mkmdSMPHz27 Just for information: crossref is looked at by a student. Hope, it is manageable ^^

@koppor thank you and that is awesome! This really needs to get done. @Toro520 I appreciate you taking a look at it!